src/smartcontracts/argSerializer.ts

```diff
@@ -158,6 +167,13 @@ export class ArgSerializer {
                 }
             } else if (value.hasExactClass(VariadicValue.ClassName)) {
                 let valueAsVariadic = <VariadicValue>value;
+                let variadicType = <VariadicType>valueAsVariadic.getType();
+
+                if (variadicType.isCounted) {
+                    const countValue = new U32Value(valueAsVariadic.getItems().length);
+                    buffers.push(self.codec.encodeTopLevel(countValue));
```

Comment:
`@theNumberOfVariadicArguments@var1@var2@var3`.

Reply:
Maybe the variadic arg handling can be pulled into a function so we have less complexity in the if/else branches.

Q: do you like the `else if {}` approach better? Since every branch returns, we could also have less indenting doing:
```
if (condition) {
  ...
  return result
}

if (condition2) {
  ...
  return result2;
}

Reply:
Indeed, fixed. Extracted handling into separate (though "nested") functions. Also re-organized branches for less indenting :+1: